src/gui/tray/NCIconWithBackgroundImage.qml

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+import QtQuick 2.15
+import Style 1.0
+
+Item {
+    id: root
+
+    property alias source: icon.source
+    property alias sourceSize: icon.sourceSize
+    property alias backgroundIconSource: backgroundImage.source
+
+    implicitWidth: backgroundImage.source !== "" ? backgroundImage.implicitWidth : icon.implicitWidth
+    implicitHeight: backgroundImage.source !== "" ? backgroundImage.implicitHeight : icon.implicitHeight
+
+    Image {
```

Comment:
I would probably get the background image being the root item and the icon image be on top of it
will save an item (better for performance) and ensure the stacking order works as intended
otherwise you should probably set a stacking order using the `z` property for both images

Reply:
@mgallien fixed, and added a screenshot into description ^^